### ISSUE TITLE
UWP build breaks when using tsconfig.json & switching between debug/release

### ISSUE THREAD

**Issue by grork** [Microsoft/TypeScript#20628]
**TypeScript Version:**  2.5

**Code**
[TSRepro.zip](https://github.com/Microsoft/TypeScript/files/1549203/TSRepro.zip)

_tl;dr:_
1. TypeScript app has a tsconfig.json
2. Build it as Release/AnyCPU
3. Build it as Debug/AnyCPU

~16,000 errors, and a 30second VS hang post build. Example error from above repro:
`1>C:\Users\dhopt\Documents\Visual Studio 2017\Projects\TSRepro\TSRepro\bld\Debug\winrtrefs\winrtrefs.d.ts(2,11): error TS2300: Build:Duplicate identifier 'Promise'.`
😿 

**Expected behavior:**
Same as VS2017 Update 4 -- when building, the other flavours are filtered/adjusted as needed so mul

**Actual behavior:**
Build failure because multiple winrtrefs.d.ts are in the module resolution paths (implicitly).

**Comment by minestarks** [comment]
@grork Just making sure of two things:

Does cleaning the build / deleting the old `winrtrefs.d.ts` work around the issue?

Also, this shouldn't be a new bug. Did this only start happening after you upgraded?

**Comment by grork** [comment]
1. Yes, deleting them **manually** did workaround the issue. However, a _clean_ didn't without extra work -- it cleaned that build flavors files, but the imports were crossing across different flavours. If I switched to the not-needed-flavour and cleaned, then switched back I believe it fixed itself up.
2. At the time of logging this issue, I thought it _was_ a regression. But after a lot of testing (in my attempt to clean it up) I _think_ I concluded it wasn't new, but something caused me to (unintentionally) build a different cpu/flav combo when I updated. I'm going to attribute that to _gremlins_.

**Comment by minestarks** [comment]
Sorry for letting this linger for so long. Managing tsconfig.json in UWP projects gets a little tricky, especially since the tsconfig.json doesn't have access to the msbuild configuration. I'd recommend doing one or both of the following:

- Define the `<WinRt2TsOutputDir>` msbuild property to dictate exactly `winrtrefs.d.ts` will go
- Use `include`/`exclude` patterns or the `files` property to avoid including the generated files under `bld`, except for the generated `winrtrefs.d.ts`. 

Right now we don't have a way for VS to do this "automatically", as in, we won't touch the contents of your tsconfig.json file. The above solutions should work pretty well though.

**Comment by grork** [comment]
Oohh, the WinRtTsOutputDir seems like a great place for me to start -- seems logical, and fits in the 'uwps build with msbuild' philosophy.
